src/hooks/contracts/useTransaction.tsx

```diff
@@ -109,14 +113,22 @@ export default function useTransaction<
       const contract = new Contract(address, abi, signer) as MyContract
       try {
         console.info('Calculating transaction gas.')
+        if (appChainId === Chains.optimism) {
+          const txReq = await contract.populateTransaction[method as string](..._params)
+          const tx = await signer.populateTransaction(txReq)
+          const result = await (
+            web3Provider as optimismSDK.L2Provider<Web3Provider>
+          ).estimateTotalGasCost(tx) // L1 + L2 fees
```

Comment:
I would assume this return the transaction COST (in wei), which is different from the gas used. Otherwise how can you compute the L1 fee? You'd have to get the current gas price on L1, multiply...etc. 

So it means that in your if statement, you return two different type of data.
If OP => transaction cost in WEI
If other => gas estimate

Reply:
have a look at the function here: https://github.com/ethereum-optimism/optimism/blob/develop/packages/sdk/src/l2-provider.ts